sale_order_project/models/sale.py

```diff
@@ -59,8 +65,34 @@ def action_create_project(self):
         project_obj = self.env['project.project']
         for order in self:
             vals = self._prepare_project_vals(order)
-            project = project_obj.create(vals)
-            order.write({
-                'project_id': project.analytic_account_id.id
-            })
-        return True
+            if order.project_template_id:
+                new_date_start = time.strftime('%Y-%m-%d')
+                new_date_end = False
+                if order.project_template_id.date_start and \
+                        order.project_template_id.date:
+                    start_date = date(*time.strptime(
+                            order.project_template_id.date_start, '%Y-%m-%d'
+                        )[:3])
+                    end_date = date(*time.strptime(
+                            order.project_template_id.date,
+                            '%Y-%m-%d')[:3])
+                    new_date_end = (datetime(*time.strptime(
+                            new_date_start,
+                            '%Y-%m-%d'
+                        )[:3])+(end_date-start_date)).strftime('%Y-%m-%d')
```

Comment:
AFAICT if the template has start and end dates, you take the duration from it and apply it to the new project.
This feature should be documented, at least in the README.
You get bonus points if you can improve the readability to make this evident 😸

Reply:
I added it to the README, with the text to datetime conversions I find it quite difficult to make the code readable. But if you have any idea of how to tackle it, I'd like to try.